src/workerd/io/actor-cache.c++

```diff
@@ -2608,49 +2608,48 @@ kj::Promise<void> ActorCache::flushImplUsingSinglePut(kj::Vector<PutBatch> putBa
 
   // See the comment in flushImplUsingTxn for why we need to construct our RPC and then wait on
   // reads before actually sending the write. The same exact logic applies here.
-  return oomCanceler.wrap(waitForPastReads().then([putBatches = kj::mv(putBatches)]() mutable {
```

Comment:
My only fear is I'm missing something important about OOM Canceler operating on `waitForPastReads`. I checked out the impl and nothing stood out, but I take you're confident that moving it to `flushImpl` is ... OK?

Reply:
Yeah, the biggest weirdness is that we don't wrap the deleteAll in oom cancelation currently. We probably could but I've let it be for now. I suspect the most effective place to do the wrap is around `flushImpl` itself. Again, not necessary for this so I've skipped it for now.